dev/integration_tests/android_host_app/app/src/main/java/io/flutter/add2app/MainActivity.java

```diff
@@ -1,14 +1,13 @@
 package io.flutter.add2app;
 
-import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
-import io.flutter.facade.Flutter;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.view.FlutterMain;
 
-public class MainActivity extends AppCompatActivity {
-
-  @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    setContentView(Flutter.createView(this, getLifecycle(), "route1"));
-  }
+public class MainActivity extends FlutterActivity {
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        FlutterMain.startInitialization(getApplicationContext());
```

Comment:
Do we need an `ensureInitializationComplete()` after this?

Reply:
You always do it on onAttach in FlutterActivityAndFragmentDelegate but only if you don't pre-warm the engine which does feel a bit implementation leaky. 

What do you think? 

Do you want the users to know the difference and just document these everywhere or make the users always do it? 

https://github.com/flutter/flutter/issues/33145 also comes into question since no equivalents exist on iOS.

Reply:
I forgot about the delegate. This is probably fine then.